test/acceptance/features/steps/steps.py

```diff
@@ -145,26 +152,35 @@ def db_instance_is_running(context, db_name):
 @when(sbr_is_applied_step)
 def sbr_is_applied(context):
     sbr_yaml = context.text
-    sbr = ServiceBindingRequest()
-    if context.__contains__("nodejs_app"):
-        application = context.nodejs_app
-        context.nodejs_app_original_generation = application.get_observed_generation()
-        context.nodejs_app_original_pod_name = application.get_running_pod_name()
+    namespace = context.namespace
+    sbr = ServiceBindingRequest(namespace.name)
+    if context.__contains__("application"):
+        application = context.application
+        context.application_original_generation = application.get_observed_generation()
+        if context.application_type.__contains__("knative"):
+            context.application_original_rev_name = application.get_rev_name()
+        else:
+            context.application_original_pod_name = application.get_running_pod_name()
```

Comment:
do we really need these here? what do we achieve with it since we overwrite the values for example in `imported_nodejs_app_is_running(context, application_name` method?

Reply:
I feel it should be here for capturing generation/observed generation, reason being as we check if application is redeployed, we would need to check these generation right before sbr is applied.